From 1f8e69eb81f05503c498e86636dd87d6f4306f6b Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 9 Apr 2021 16:49:21 +0200
Subject: [PATCH] lineedit: fix tab completion with equal sign

Fix tab completion for the path when equal sign (=) is used. For
example: dd if=/dev/ze<tab>

Signed-off-by: Natanael Copa <ncopa@alpinelinux.org>
---
 libbb/lineedit.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libbb/lineedit.c b/libbb/lineedit.c
index b0adcf140..2cae4711a 100644
--- a/libbb/lineedit.c
+++ b/libbb/lineedit.c
@@ -1071,7 +1071,7 @@ static NOINLINE int build_match_prefix(char *match_buf)
 		continue;
 	for (--i; i >= 0; i--) {
 		int cur = int_buf[i];
-		if (cur == ' ' || cur == '<' || cur == '>' || cur == '|' || cur == '&') {
+		if (cur == ' ' || cur == '<' || cur == '>' || cur == '|' || cur == '&' || cur == '=') {
 			remove_chunk(int_buf, 0, i + 1);
 			break;
 		}
-- 
2.31.1

